src/agenda/index.js

```diff
@@ -139,13 +138,21 @@ export default class AgendaView extends Component {
     this.state.scrollY.removeAllListeners();
   }
 
-  UNSAFE_componentWillReceiveProps(props) {
-    if (props.items) {
-      this.setState({
-        firstResevationLoad: false
-      });
+  UNSAFE_componentWillReceiveProps(nextProps) {
+    const nextState = {};
+    if (nextProps.items) {
+      nextState.firstResevationLoad = false;
+    }
+    if (
+      nextProps.selected &&
+      !dateutils.sameDate(nextProps.selected, this.props.selected)
+    ) {
+      nextState.selectedDay = parseDate(nextProps.selected);
+    }
+    if (Object.keys(nextState).length) {
+      this.setState(nextState);
     } else {
```

Comment:
I think the 'else' statement is misplaced. It should be attached to the `if(props.items)`, no?

Reply:
Yes, you're right